### ISSUE TITLE
Refine and add more param to the LSTM/MTNetForecaster

### ISSUE THREAD

**Issue by TheaperDeng** [intel-analytics/analytics-zoo#2405]
Some of the param name in MTNet and LSTM Forecaster is confusing.
Some of the param of MTNet and LSTM is not exposed to the Forecaster level.

**Comment by TheaperDeng** [comment]
Original name | Modified name | Description | Default value | Corresponding Model variable | Note for changing
-- | -- | -- | -- | -- | --
target_dim | target_dim | Specify the number of variables we want to forecast. i.e. the the dimension of model output feature. | 1 | self.output_dim |  
feature_dim | feature_dim | Specify the number of variables we have in the input data. i.e. the the dimension of model input feature. | 1 | self.feature_num |  
ar_window_size | ar_window_size | Specify the auto regression window size in MTNet. | 1 | self.ar_window |  
metric | metric | Specify the metric for validation and evaluation. | MSE | self.metrics |  
uncertainty | uncertainty | Specify whether the forecaster can perform the calculation of uncertainty. | False | self.mc |  
cnn_kernel_size | cnn_height | Specify convolutional layer filter height in MTNet. | 1 | self.cnn_height | only the height can be set.
lb_long_steps | long_series_num | Specify the number of long-term historical data series. | 1 | self.long_num |  
lb_long_stepsize | series_length | Specify the length of long-term historical data series, which is equal to the length of short-term data series. | 1 | self.time_step | for long and short this is the same value.
None | cnn_hid_size | Specify the number of cnn filters in the encoder. | 32 | self.cnn_hid_size |  
None | rnn_hid_sizes | Specify the dim of RNN output in the encoder. It should be a list. | [16,32] | self.rnn_hid_sizes |  
None | lr | Specify the learning rate. | 0.001 | self.lr |  
None | loss | Specify the loss function. | mae | - |  
None | optimizer | Specify the optimizer in training. | adam | - |  
None | cnn_dropout | Specify the fraction of the input units to drop for the cnn in encoder. | 0.2 | self.dropout |  
None | rnn_dropout | Specify the fraction of the input units to drop for the rnn in encoder. | 0.2 | self.dropout | For now, cnn and rnn are using the same p for dropout

Here is a changing plan to the MTNet model.

**Comment by TheaperDeng** [comment]
Original name | Modified name | Description | Default value | Corresponding Model variable | Note for changing
-- | -- | -- | -- | -- | --
target_dim | target_dim | Specify the number of variables we want to forecast. i.e. the the dimension of model output feature. | 1 |   |  
feature_dim | feature_dim | Specify the number of variables we have in the input data. i.e. the the dimension of model input feature. | 1 |   |  
lstm_1_units | lstm_1_units | Specify the dimensionality of the output space for 1st LSTM layer. | 16 |   |  
dropout_1 | dropout_1 | Specify the dimensionality of the output space for 1st LSTM layer. | 0.2 |   |  
lstm_2_units | lstm_2_units | Specify the dimensionality of the output space for 2nd LSTM layer. | 8 |   |  
lstm_2_units | lstm_2_units | Specify the dimensionality of the output space for 2nd LSTM layer. | 0.2 |   |  
metric | metric | Specify the dimensionality of the output space for 2nd LSTM layer. | mean_squared_error |   |  
lr | lr | Specify the learning rate. | 0.001 |   |  
uncertainty | uncertainty | Specify whether the forecaster can perform the calculation of uncertainty. | False |   |  
None | loss | Specify the loss function. | mse |   |  
None | optimizer | Specify the optimizer using in the training process. | RMSprop |   |  

Here is a changing plan to the LSTM model.
